packages/mobile/src/pages/chat/edit-group-page.tsx

```diff
@@ -97,34 +150,280 @@ export function EditGroupPage(props: EditGroupProp) {
         ]}
       >
         <Header
-          title={'Edit Group Title'}
+          title={t('Edit Group')}
           headerLeft={headerLeft}
           headerRight={headerRight}
           headerStyle={{
             backgroundColor: headerColor,
           }}
         />
       </View>
-      <SafeAreaView>
-        <View style={styles.content}>
-          <GroupAvatar style={styles.avatar} iconSize={'2xl'} />
-          <TextButton title="Edit Avatar" onPress={pickAvatar} />
-          <Input
-            value={title}
-            onChangeText={(title) => setTitle(title)}
-            autoCorrect={false}
-            autoFocus={true}
-            autoCapitalize="sentences"
-            style={{
-              width: '70%',
+
+      <View style={styles.content}>
+        <GroupAvatar
+          fileId={conversation.fileId}
+          fileKey={hasGroupImage ? groupAvatarPayloadKey : undefined}
+          targetDrive={ChatDrive}
+          previewThumbnail={conversation.fileMetadata.appData.previewThumbnail}
+          imageStyle={{
+            width: 81,
+            height: 81,
+          }}
+          style={styles.avatar}
+          iconSize={'2xl'}
+        />
+        {/* <TextButton title="Edit Avatar" onPress={pickAvatar} /> //TODO: When BE fixes appendPayload, this comes back :) */}
+        <Input
+          value={title}
+          onChangeText={(title) => setTitle(title)}
+          autoCorrect={false}
+          autoFocus={true}
+          autoCapitalize="sentences"
+          style={{
+            width: '70%',
+          }}
+        />
+      </View>
+      <View
+        style={{
+          flexDirection: 'row',
+          justifyContent: 'space-between',
+          alignItems: 'center',
+          marginTop: 12,
+          paddingHorizontal: 16,
+        }}
+      >
+        <Text
+          style={{
+            fontSize: 22,
+            fontWeight: '500',
+          }}
+        >
+          Recipients
+        </Text>
+        <TextButton title="Add Members" onPress={() => setVisible(true)} />
+      </View>
+      <ScrollView
+        style={{
+          flex: 1,
+        }}
+      >
+        <RenderRecipientTile recipient={identity as string} isMe={true} />
+        {recipients?.map((recipient) => (
+          <RenderRecipientTile key={recipient} recipient={recipient} isMe={false} />
+        ))}
+        {newRecipients.map((recipient) => (
+          <RenderRecipientTile
+            key={recipient}
+            recipient={recipient}
+            isMe={false}
+            newRecipient={newRecipients.includes(recipient)}
+            onRemove={(odinId) => {
```

Comment:
Let move this one into the a useCallback. And you should even be able to have that one only depend on setNewRecipients, something like this:

```
useCallback(() => 
  setNewRecipients(currentRecipients => currentRecipients.filter((x) => x !== odinId))
), [setNewRecipients]);
```